editoast/src/modelsv2/infra.rs

```diff
@@ -125,65 +122,92 @@ impl Infra {
+
+            // Disable triggers to speed up the cloning
+            sql_query("ALTER TABLE infra_object_signal DISABLE TRIGGER search_signal__ins_trig").execute(conn).await?;
+            sql_query("ALTER TABLE infra_object_track_section DISABLE TRIGGER search_track__ins_trig").execute(conn).await?;
+            sql_query("ALTER TABLE infra_object_operational_point DISABLE TRIGGER search_operational_point__ins_trig").execute(conn).await?;
+
+            // Fill cloned infra with data
+            for object in ObjectType::iter() {
+                let model_table = get_table(&object);
+                sql_query(format!(
+                    "INSERT INTO {model_table}(obj_id,data,infra_id) SELECT obj_id,data,$1 FROM {model_table} WHERE infra_id = $2"
+                ))
+                .bind::<BigInt, _>(cloned_infra.id)
+                .bind::<BigInt, _>(self.id)
+                .execute(conn).await?;
+
+                if let Some(layer_table) = get_geometry_layer_table(&object) {
+                    let layer_table = layer_table.to_string();
+                    let sql = if object != ObjectType::Signal {
+                        format!("INSERT INTO {layer_table}(obj_id,geographic,infra_id) SELECT obj_id,geographic,$1 FROM {layer_table} WHERE infra_id=$2")
+                    } else {
+                        format!("INSERT INTO {layer_table}(obj_id,geographic,infra_id, angle_geo, signaling_system, sprite) 
+                                    SELECT obj_id,geographic,$1,angle_geo, signaling_system, sprite FROM {layer_table} WHERE infra_id = $2")
+                    };
+
+                    sql_query(sql)
+                        .bind::<BigInt, _>(cloned_infra.id)
+                        .bind::<BigInt, _>(self.id)
+                        .execute(conn)
+                        .await?;
+                }
+            }
 
-                let layer = sql_query(sql)
+            // Re-enable triggers to speed up the cloning
+            sql_query("ALTER TABLE infra_object_signal ENABLE TRIGGER search_signal__ins_trig").execute(conn).await?;
+            sql_query("ALTER TABLE infra_object_track_section ENABLE TRIGGER search_track__ins_trig").execute(conn).await?;
+            sql_query("ALTER TABLE infra_object_operational_point ENABLE TRIGGER search_operational_point__ins_trig").execute(conn).await?;
+
+            // Fill search tables
+            sql_query("INSERT INTO search_signal(id, label, line_name, infra_id, obj_id, signaling_systems, settings, line_code) 
+                        SELECT signal.id, label, line_name, $1, search_signal.obj_id, signaling_systems, settings, line_code FROM search_signal 
+                        JOIN infra_object_signal AS signal ON search_signal.obj_id = signal.obj_id and signal.infra_id = $1
+                        WHERE search_signal.infra_id = $2")
+                .bind::<BigInt, _>(cloned_infra.id)
+                .bind::<BigInt, _>(self.id)
+                .execute(conn).await?;
```

Comment:
Could you add a comment in `search::objects` that these queries will require some modification should any changes be made to their table schema?

Reply:
Done